bundles/org.eclipse.ui/src/org/eclipse/ui/internal/SearchDecoration.java

```diff
@@ -62,20 +67,33 @@ public static boolean validateRegex(String regex, ControlDecoration targetDecora
 	 * @return The appropriate error message if the regex is invalid or an empty
 	 *         string if the regex is valid.
 	 */
-	private static String getValidationError(String regex) {
+	private static String getValidationError(String regex, GC gc) {
 		try {
 			Pattern.compile(regex);
 			return ""; //$NON-NLS-1$
 		} catch (PatternSyntaxException e) {
-			String message = e.getLocalizedMessage();
+			String description = e.getDescription();
+			int errorIndex = e.getIndex();
```

Comment:
According to the JavaDoc, `getIndex()` may return -1. In that case you should only show the error message (no extra lines)

Reply:
Done